dask_planner/src/sql/optimizer.rs

```diff
@@ -40,10 +40,14 @@ impl DaskSqlOptimizer {
     ) -> Result<LogicalPlan, DataFusionError> {
         let mut resulting_plan: LogicalPlan = plan;
         for optimization in &self.optimizations {
-            match optimization.optimize(&resulting_plan, &OptimizerConfig::new()) {
+            match optimization.optimize(&resulting_plan, &mut OptimizerConfig::new()) {
```

Comment:
Curious why this needs to be `mut` now

Reply:
There is an Id generator in the config strict now, for generating aliases